### ISSUE TITLE
PropertyGrid CreateDefaultEditor exception

### ISSUE THREAD

**Issue by xceedsoftware** [xceedsoftware/wpftoolkit#439]
<b>ThisWillFail[CodePlex]</b> <br />in version 1.7 PropertyGridUtilities.cs, CreateDefaultEditor function, line 195:else if( propertyItem.PropertyType.IsEnum )it gives an exception if PropertyType is null. This means that the propertygrid ends up being empty.For users who try to make it
 work for the first time, it would be easier if it would use a text box, because this would show them that they are on to something, and then they could improve their code afterwards if they need it to use another editor. else if( propertyItem.PropertyType
 == null ) editor = new TextBoxEditor();in the beginning of the list of else if's would do this.

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />should be fixed in v2.5 and up.

**Comment by xceedsoftware** [comment]
<b>emartin[CodePlex]</b> <br />I got another question for you.
Would something like an ItemsControl API containing an ItemsSource property that support a IEnumerable of custom object where each object represent a property would have been a much easier path for your needs ?

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Can you reproduce this bug in v1.8 ? If yes can you provide a sample of this bug ?

**Comment by xceedsoftware** [comment]
<b>ThisWillFail[CodePlex]</b> <br />Short answer: No. 

I am creating an application where the list of properties in the propertygrid will be dynamic. I host the list of properties in a property class.


In order to get propertygrid to read my properties right I have to create a class that inherits from PropertyDescriptor. I only do this to make it work in propertygrid, so I only populate what I have to to make this work. At first I just populated displayname
 and value. When I got a blank propertygrid I assumed I did something wrong in the xaml. It was not until I downloaded the wpftoolkit source and debugged through it I saw that it was the missing PropertyType that made it empty. If I had gotten text boxes for
 this I would have seen that I was on the right path, and I could have continued adding PropertyType to get the right controls. My intention was always to populate PropertyType, but at the time I didn't even know if using the PropertyDescriptor class would
 work, so I didn't want to fully populate it before testing it. 

I don't need the change that I am suggesting, I have moved beyond this point now. I am just suggesting it so it will be easier for the next one who tries. Many programmers create user interface through trial and error instead of reading a book and doing it
 right the first time. This change will make it easier for people who work through trial and error.
